README.md
# Mazy boi

Mazy boi is a rpg base maze game

## Installation

Python 3.7 is needed to play this game. [Pyhon 3.7](https://www.python.org/downloads/) to install Python.

## Usage

```python
python game.py
```


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Prerelease Version]

## [0.0.1] - 2019-10-03
### Added
 - Created CHANGELOG.md
 - Created Task Todo

### Changed
- Updated README.md

## [0.0.2] - 2019-10-03
### Changed
- Updated README.md
- Updated CHANGELOG.md

## [0.0.3] - 2019-10-03
### Changed
- Added task in To do list and marked Create a README.md, Create CHANGELOG.md
and Create a To do list done

## [0.0.4] - 2019-10-24
### Added
 - Created properties.py to contain nested dictionaries

### Changed
 - Update to task.todo

## [0.0.5] - 2019-10-24
### Added
 - Created menu.py that will be the base of the text base adventure game for
 interaction

### Changed
 - Update to task.todo


test.py
# Josh Lim
# Comp Sci 30 P4
# 10/24/2019
# For testing stuff
import properties


print("\nInventory")
# Prints out the content of the weapons nested dictionary
properties.nestedDictPrinter(properties.inventory)
print("\nMobs")
# Prints out the content of the mobs nested dictionary
properties.nestedDictPrinter(properties.mobs)
print("\nCharacter")
# Prints out the content of the character nested dictionary
properties.nestedDictPrinter(properties.character)
print("\nLocations")
# Prints out the content of the locations nested dictionary
properties.nestedDictPrinter(properties.locations)


menu.py
# Josh Lim
# Comp Sci 30 P4
# 10/29/2019
# Simple action menu for Game
import locations


def menuDesc(menuName, key):
    """Function that prints out a description for the path chosen"""
    print(f"Path {key}:")
    print(f"\t{menuName[key]}")

def pathMenu(menuName):
    """Function that will take a dictionary, menuName, and converts it to a
    text-based menu
    """
    # Prints keys
    for key in menuName:
        print("\tPath " + key)
    # Input for choosing the menu
    choice = input("What path shall thoust choose? ")
    # Creates a list of keys for error checking
    choices = list(menuName)
    # Loop that checks if the user enter an invalid input
    while str(choice) not in choices:
        print(f"I got, {str(choices)[1:-1]} watchu want?")
        choice = input("")
        # Repeats input for choosing the menu
    for key in choices:
        if choice == key:
            # Prints out users choice
            print(f"You have chose Path {key}")
            menuDesc(menuName, choice)
    choice = "path" + choice
    return choice

def pathChoose():
    """Function to determine the next path the player will take"""
    print(pathMenu)
    for path in locations.paths:
        if str(path) == pathMenu:
            print(pathMenu)


pseudocode.py
# Title screen will show first
# Option of play and how to play

# When clicked how to play a instruction will be printed on screen

"""When clicked play, the game will commence and the player will find he/she
in a maze and will have to go through the maze in order to get out, player
will then have the choice to go through path a, b or c"""

# Path a
"""Randomly generate a level 0 mob"""

"""Player will now have two choices to go through path aa, ab or ac"""
# Path b

# Path c

# Path aa
"""Randomly generate a level 1 mob"""

"""Player will now have the choice to go to path aaa, aab, or aac"""
# Path ab

# Path ac

# Path ba

# Path bb

# Path bc

# Path ca

# Path cb

# Path cc

# Path aaa
"""Randomly generates a level 2 mob"""

"""Player will have a choice to go to path aaaa, aaab or aaac"""

# Path aab

# Path aac

# Path aba

# Path abb

# Path abc

# Path aca

# Path acb

# Path acc

# Path aaaa
"""Randomly generates a level 3 mob"""

"""Player will now have the choice to go to path aaaaa, aaaab, or aaaac"""
# Path aaaaa
"""Player will fight mob boss"""

"""Game is completed once boss is defeated"""


informations.py
Weapons = {
    "Stick": {
        "ATTACK": 3,
        "DURABILITY": 5,
        },
    "Iron Sword": {
        "ATTACK": 15,
        "DURABILITY": 20,
        "SHARPNESS": 10,
        }
    }

for weapon, info in Weapons.items():
    print(weapon)
    if 'ATTACK' in info.keys():
        print(f"\tAttack: {info['ATTACK']}")
    if 'DURABILITY' in info.keys():
        print(f"\tDurability: {info['DURABILITY']}")
    if 'SHARPNESS' in info.keys():
        print(f"\tSharpness: {info['SHARPNESS']}")

Mobs = {
    "Spider": {
        "ATTACK": 3,
        "HP": 10,
        "VITALITY": 10,
        "DEFENSE": 10,
        "LUCK": 0,
        }
    }

for mob, info in Mobs.items():
    print(mob)
    if 'ATTACK' in info.keys():
        print(f"\tAttack: {info['ATTACK']}")
    if 'HP' in info.keys():
        print(f"\tHP: {info['HP']}")
    if 'VITALITY' in info.keys():
        print(f"\tVitality: {info['VITALITY']}")
    if 'DEFENSE' in info.keys():
        print(f"\tDefense: {info['DEFENSE']}")
    if 'LUCK' in info.keys():
        print(f"\tLuck: {info['LUCK']}")

hero = {
    "inventory":["map", "flashlight"],
    "stats": {
        "ATTACK": 10,
        "HP": 100,
        "VITALITY": 10,
        "DEFENSE": 10,
        "LUCK": 0,
        }
}

print(hero["inventory"])
for character, info in hero.items():
    print(character)
    print(info)


properties.py
# Josh Lim
# Comp Sci 30 P4
# 10/24/2019
# Nested dictionaries for game properties


def nestedDictPrinter(dictName):
    """Prints out content in a nested dictionary with up to 2 layers,
    takes dictName as the dictionary name.
    """
    for dictName, key in dictName.items():
        """loops through the keys in the nested dictionary, dictName, and then
        prints the nested dictionary names along with its values.
        """
        print(dictName.title() + ":")  # prints nested dictionary name
        for items in key:
            """loops through nested dictionary key values and and prints the
            key along with the value with a tab before.
            """
            print(f"\t{items.title()} - {key[items]}")
# Nested dictionary containing information for inventory
inventory = {
    "stick": {
        "ATTACK": 3,
        "DURABILITY": 5,
        },
    "iron sword": {
        "ATTACK": 15,
        "DURABILITY": 20,
        "SHARPNESS": 10,
        }
    }
# Nested dictionary containing information for mobs
mobs = {
    "spider": {
        "ATTACK": 3,
        "HIT POINTS": 10,
        "VITALITY": 10,
        "DEFENSE": 10,
        "LUCK": 0,
        }
    }
# Nested dictionary containing information for character
character = {
    "stats": {
        "ATTACK": 10,
        "HIT POINTS": 100,
        "VITALITY": 10,
        "DEFENSE": 10,
        "LUCK": 0,
        }
}
# Nested dictionary containing information for locations
locations = {
    "path 1": {
        "description": "You have encountered a level 1 mob!",
        },
    "path 2": {
        "description": "You are greeted with 2 paths, which will you choose?",
        },
    "path 3": {
        "description": "You find yourself in a forest, what will you do?",
    }
}
print("\nInventory")
# Prints out the content of the weapons nested dictionary
nestedDictPrinter(inventory)
print("\nMobs")
# Prints out the content of the mobs nested dictionary
nestedDictPrinter(mobs)
print("\nCharacter")
# Prints out the content of the character nested dictionary
nestedDictPrinter(character)
print("\nLocations")
# Prints out the content of the locations nested dictionary
nestedDictPrinter(locations)
